utils/forms.py

```diff
@@ -38,23 +42,27 @@ def clean(self, value):
         value = super(HtmlCleaningCharField, self).clean(value)
         if is_shouting(value):
             raise forms.ValidationError('Please moderate the amount of upper case characters in your post...')
-        try:
-            return clean_html(value)
-        except (HTMLParseError, UnicodeEncodeError):
-            raise forms.ValidationError('The text you submitted is badly formed HTML, please fix it')
+        return clean_html(value)
 
 
 class TagField(forms.CharField):
     """ Gets the value of tags as a single string (with tags separated by spaces or commas) and cleans it to a set of
     unique tag strings """
-    def clean(self, value):
-        tags = clean_and_split_tags(str(value))
-        if len(tags) < 3:
-            raise forms.ValidationError('You should add at least 3 different tags. Tags must be separated by spaces '
-                                        '(and/or commas).')
-        elif len(tags) > 30:
-            raise forms.ValidationError('There can be maximum 30 tags, please select the most relevant ones!')
-        return tags
+
+    def __init__(self, **kwargs):
+        super(TagField, self).__init__(**kwargs)
+        self.validators.append(
+            validators.MinLengthValidator(3, 'You should add at least 3 different tags. Tags must be separated by '
+                                             'spaces'))
```

Comment:
after discussion with @ffont we removed commas from this error message, now we just insist on using spaces